File: Backend/src/main/java/dev/group21/rescmeal/model/Comment.java
package dev.group21.rescmeal.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "comment")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "client_id")
    private Client client;

    @ManyToOne
    @JoinColumn(name = "carrier_id")
    private Carrier carrier;

    @ManyToOne
    @JoinColumn(name = "purchase_id")
    private Purchase purchase;

    @Column(name = "description", columnDefinition = "text")
    private String description;
}

File: Frontend/src/types/Comment.ts
import { Carrier } from "./Carrier";
import { Client } from "./Client";
import { Purchase } from "./Purchase";

export interface Comment{
    id?: number;
    client: Client;
    carrier: Carrier;
    description: string;
    purchase: Purchase;
}

export interface CommentErrors{
    description: string;
}